File: tests/eval/java/seen/CleanLoggedRethrow.java
public class CleanLoggedRethrow {
    String read(java.io.BufferedReader r) throws java.io.IOException {
        try { return r.readLine(); }
        catch (java.io.IOException e) {   // NEAR-MISS: looks like a swallow, but logs and rethrows
            System.err.println(e);
            throw e;
        }
    }
}

File: tests/eval/csharp/seen/clean_logged_rethrow.cs
public class Loader {
    public string Read(string path) {
        try { return System.IO.File.ReadAllText(path); }
        catch (System.Exception e) {   // NEAR-MISS: looks like a swallow, but logs and rethrows
            System.Console.Error.WriteLine(e);
            throw;
        }
    }
}
